### PR TITLE
Support Juju 3.4

### PR DESCRIPTION
**Pull request by rgildein**
Switching CI run with Juju 3.4 on Juju 3.4 controller. To be able to support Juju 3.4, we need to get rid of nagios, however one test depend on it. I decide to switch to cos-proxy, since only think we need is to be able to create monitors relation.

Add workflow concurrency to avoid running multiple workflows on a single PR. Like this if new commit arrived the previous run will be canceled by new run.

~~Blocked until [#655](https://github.com/openstack-charmers/zaza/pull/655) is merged.~~

### CODE REVIEW

File: tests/functional/requirements.txt
Comment (Pjack) [inline tests/functional/requirements.txt:1]: Could we leave a comment or issue link to indicate when we can use official repository again? thanks!

**Review by Pjack — CHANGES_REQUESTED** [review]

**Review by dashmage — APPROVED** [review]
LGTM, thanks! Hopefully, the upstream zaza PR gets merged soon and we can use it directly.

File: .github/workflows/check.yaml
Comment (dashmage) [inline .github/workflows/check.yaml:37]:
```suggestion
      zaza-yaml: "LS0tCm1vZGVsX3NldHRpbmdzOgogIGltYWdlLXN0cmVhbTogcmVsZWFzZWQKcmVnaW9uOiBwcm9kc3RhY2s2CmNsb3VkOiBidWlsZGVyLWNsb3VkCmNyZWRlbnRpYWw6IGJ1aWxkZXItY2xvdWQtY3JlZAo="
```
From the failed func test, I found that we were using the wrong zaza cloud and credential config (specific for 2.9). I've changed the base64 encoded string to reflect the names for the juju 3.x setup.

**Review by dashmage — CHANGES_REQUESTED** [review]
Found something after looking at the failed func test.

File: tests/functional/requirements.txt
Comment (rgildein) [reply tests/functional/requirements.txt:1]: Check description, "Blocked until https://github.com/openstack-charmers/zaza/pull/655 is merged." I do not want to merge this until that PR is not merged, what should be soon.

**Review by dashmage — APPROVED** [review]
LGTM now. Good work with the troubleshooting for the cos-proxy replacement :smile:

**Comment by dashmage** [comment]
Just a heads-up - the last run failed with a machine error and the zaza functest run command has the `--keep-faulty-model` flag added on. So every time this fails, there'll be a large model left behind in our PS6 env in need of manual cleanup.

**Comment by rgildein** [comment]
Switching to `return-code` from `Code` to fix `test_20_create_instance` test, however, I could not figure out why `test_25_test_connectivity` failed.

**Review by aieri — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]

**Comment by rgildein** [comment]
@Pjack @jneo8 @aieri Do you think it's ok to merge this even if one test is failing? I reported an issue, and I do not think it's related with Juju 3.4.

**Comment by aieri** [comment]
@rgildein I think it's ok to merge. There are no practical changes in the charm code and we can deal with the functional test failure separately

**Comment by rgildein** [comment]
> Please don't merge with failing tests, unless we know for sure that the failures are completely unrelated. This was a real test failure caused by switching the CI to juju 3.4. See #27 for the fix for the root cause of the `test_25_test_connectivity` failure.

The functional test was failing even before this change, so I saw no reason to block this change.

**Pull request merged**
